website/source/docs/providers/aws/r/s3_bucket_notification.html.markdown

```diff
@@ -0,0 +1,170 @@
+---
+layout: "aws"
+page_title: "AWS: aws_s3_bucket_notification"
+side_bar_current: "docs-aws-resource-s3-bucket-notification"
+description: |-
+  Provides a S3 bucket notification resource.
+---
+
+# aws\_s3\_bucket\_notification
+
+Provides a S3 bucket notification resource.
+
+## Example Usage
+
+### Add notification configuration to SNS Topic
+
+```
+resource "aws_sns_topic" "topic" {
+    name = "s3-event-notification-topic"
+    policy = <<POLICY
+{
+    "Version":"2012-10-17",
+    "Statement":[{
+        "Effect": "Allow",
+        "Principal": {"AWS":"*"},
+        "Action": "SNS:Publish",
+        "Resource": "arn:aws:sns:*:*:terraform-test-topic",
+        "Condition":{
+            "ArnLike":{"aws:SourceArn":"${aws_s3_bucket.bucket.arn}"}
+        }
+    }]
+}
+POLICY
+}
+
+resource "aws_s3_bucket" "bucket" {
+	bucket = "your_bucket_name"
+}
+
+resource "aws_s3_bucket_notification" "bucket_notification" {
+	bucket = "${aws_s3_bucket.bucket.id}"
+	topic {
+		topic = "${aws_sns_topic.topic.arn}"
+		events = ["s3:ObjectCreated:*"]
+		filter_suffix = ".log"
+	}
+}
+```
+
+### Add notification configuration to SQS Queue
+
+```
+resource "aws_sqs_queue" "queue" {
+    name = "s3-event-notification-queue"
+    policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": "*",
+      "Action": "sqs:SendMessage",
+	  "Resource": "arn:aws:sqs:*:*:s3-event-notification-queue",
+      "Condition": {
+        "ArnEquals": { "aws:SourceArn": "${aws_s3_bucket.bucket.arn}" }
+      }
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_s3_bucket" "bucket" {
+	bucket = "your_bucket_name"
+}
+
+resource "aws_s3_bucket_notification" "bucket_notification" {
+	bucket = "${aws_s3_bucket.bucket.id}"
+	queue {
+		queue = "${aws_sqs_queue.queue.arn}"
+		events = ["s3:ObjectCreated:*"]
+		filter_suffix = ".log"
+	}
+}
+```
+
+### Add notification configuration to Lambda Function
+
+```
+resource "aws_iam_role" "iam_for_lambda" {
```

Comment:
I can't reproduce fail put notification configuration.
Please tell me what kind of AWS returned error code.
Following error?

* Error Code: InvalidArgument
* Error Message: Unable to validate the following destination configurations